### ISSUE TITLE
Exclusions aren't applied to the first directory found for each browse.path entry

### ISSUE THREAD

**Issue by akhodeir** [microsoft/vscode-cpptools#10205]
### Environment

- OS and Version: Mac OS
- VS Code Version: 1.73.1
- C/C++ Extension Version: Microsoft extensions (C/C++ and C/C++ extension back)
- Other extensions you installed (and if the issue persists after disabling them):  CodeLLDB
- If using SSH remote, specify OS of remote machine:  No
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

### Bug Summary and Steps to Reproduce

Bug Summary:
I created a workspace with many folder. Inside each folder there exists many files.

I build this settings.json:

{
      "files.associations": {
            "memory": "c",
            "variant": "c"
      },
      "search.exclude": {
            "**/firmware": true,
            "**/cellular" : true,
            "**/common":true,
            "**/embedded_headers" : true,
            "**/output" : true,
            "**/platform_sw":true,
      },
      "files.exclude": {
            "**/*.unt": true,
            "**/*.c":false,
      },
      "C_Cpp.exclusionPolicy": "checkFilesAndFolders",
      "C_Cpp.intelliSenseEngine": "Default",
}

the **"search.exclude"** entry contains all folder in the project.

Steps to reproduce:
see above.
I also reset the IntelliSense DB.

### Expected behavior

When I reset IntelliSense DB, I still see its progress as multiple files. I expected no files to be included since all folders where added in the search exclude entry.

I expected that IntelliSense wouldn't find any function declaration, when I right click on a function and press "go to definition"

### Code sample and Logs

```shell
c_cpp_properties.json doesn't exist in my project.
```

**Comment by sean-mcmanus** [comment]
Our extension doesn't look at the search.exclude feature, because we consider that setting to only apply to VS Code's search. You should set C_Cpp.files.exclude instead.

**Comment by akhodeir** [comment]
Thanks for the reply.
I added the entry to the settings.json

      "C_Cpp.files.exclude": {

            "**/firmware_pmu": true,
            "**/cellular" : true,
            "**/tools": true,
            "**/common":true,
            "**/embedded_headers" : true,
            "**/firmware_bb":true,
            "**/output" : true,
            "**/platform":true,
            "**/vinyl":true,        
            "**/.vscode": true,
            "**/.vs": true
      },
      "C_Cpp.codeAnalysis.exclude": {
            "**/firmware_pmu": true,
            "**/cellular" : true,
            "**/tools": true,
            "**/common":true,
            "**/embedded_headers" : true,
            "**/firmware_bb":true,
            "**/output" : true,
            "**/platform":true,
            "**/vinyl":true,        
            "**/.vscode": true,
            "**/.vs": true
      },
when I reset the intellisense database, I still see it discovering all files in the project. (all folders in the project are excluded for testing the extension)
![IMG-3221](https://user-images.githubusercontent.com/42145977/205013793-ebdecbae-54e3-4501-a9ae-a711cd96e8f2.jpg)

**Comment by sean-mcmanus** [comment]
Can you set C_Cpp.loggingLevel to "Debug" and look at the "C/C++" logging to see which folder we're being processed, e.g.
```
Discovering files...
  Processing folder (recursive): 
```

**Comment by akhodeir** [comment]
Thanks for the hints.

I enabled the debug mode. here is the output:

`
  Processing folder (recursive): /Library/Developer/CommandLineTools/SDKs/******
  Processing folder (recursive): /Library/Developer/CommandLineTools/SDKs/******
  Discovering files: 41383 file(s) processed
  0 file(s) removed from database
Done discovering files.
Populating include completion cache.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.

`

I tried to disable them via C_Cpp.codeAnalysis.exclude. ---> "/Library/Developer/CommandLineTools/SDKs/*":true
but they are still processed

**Comment by sean-mcmanus** [comment]
It appears to be a bug. The files.exclude is not applied to the root folder if it's in the browse.path. You can work around this bug via setting `    "C_Cpp.exclusionPolicy": "checkFilesAndFolders"`.

**Comment by sean-mcmanus** [comment]
But if you have "C_Cpp.exclusionPolicy": "checkFilesAndFolders" set already, then I'm not sure why the exclusion wouldn't be working (it works for me). You could potentially set ` "C_Cpp.loggingLevel": "8",` temporarily to get more exclusion info.

**Comment by sean-mcmanus** [comment]
The fix is available with 1.14.0 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.14.0
